tests/uptane_test.cc

```diff
@@ -312,35 +335,213 @@ TEST(SotaUptaneClientTest, device_register_twice) {
+  conf.provision.provision_path = uptane_test_dir + "/cred.zip";
+
+  std::string test_name1, test_name2;
+  {
+    FSStorage storage(conf);
+    Uptane::Repository uptane(conf, storage);
+    EXPECT_TRUE(uptane.initialize());
+
+    EXPECT_TRUE(boost::filesystem::exists(device_path));
+    test_name1 = Utils::readFile(device_path);
+    EXPECT_NE(test_name1, "");
+  }
+
+  // Make sure a new name is generated if the config does not specify a name and
+  // there is no device_id file.
+  {
+    Utils::copyDir("tests/test_data", uptane_test_dir);
+    boost::filesystem::remove(device_path);
+    conf.uptane.device_id = "";
+
+    FSStorage storage(conf);
+    Uptane::Repository uptane(conf, storage);
+    EXPECT_TRUE(uptane.initialize());
+
+    EXPECT_TRUE(boost::filesystem::exists(device_path));
+    test_name2 = Utils::readFile(device_path);
+    EXPECT_NE(test_name2, test_name1);
+  }
+  // If the device_id is cleared in the config, but the file is still present,
+  // re-initializing the config should still read the device_id from file.
+  {
+    conf.uptane.device_id = "";
+    FSStorage storage(conf);
+    Uptane::Repository uptane(conf, storage);
+    EXPECT_TRUE(uptane.initialize());
+
+    EXPECT_TRUE(boost::filesystem::exists(device_path));
+    EXPECT_EQ(Utils::readFile(device_path), test_name2);
+  }
+
+  // If the device_id file is removed, but the field is still present in the
+  // config, re-initializing the config should still read the device_id from
+  // config.
+  {
+    Utils::copyDir("tests/test_data", uptane_test_dir);
+    conf.uptane.device_id = test_name2;
+
+    FSStorage storage(conf);
+    Uptane::Repository uptane(conf, storage);
+    EXPECT_TRUE(uptane.initialize());
+
+    EXPECT_TRUE(boost::filesystem::exists(device_path));
+    EXPECT_EQ(Utils::readFile(device_path), test_name2);
+  }
+
+  boost::filesystem::remove_all(uptane_test_dir);
+}
+
+TEST(SotaUptaneClientTest, initialize_fail) {
   Config conf("tests/config_tests_prov.toml");
+  Utils::copyDir("tests/test_data", uptane_test_dir);
+  conf.uptane.metadata_path = "tests/";
+  conf.uptane.repo_server = tls_server + "/director";
+  conf.tls.certificates_directory = uptane_test_dir + "/certs";
+
+  conf.uptane.repo_server = tls_server + "/repo";
+  conf.tls.server = tls_server;
+
+  conf.uptane.primary_ecu_serial = "testecuserial";
+  conf.uptane.private_key_path = "private.key";
+  conf.uptane.public_key_path = "public.key";
 
   boost::filesystem::remove(conf.tls.certificates_directory / conf.tls.client_certificate);
   boost::filesystem::remove(conf.tls.certificates_directory / conf.tls.ca_file);
+  boost::filesystem::remove(conf.tls.certificates_directory / conf.tls.pkey_file);
+  boost::filesystem::remove(conf.tls.certificates_directory / "bootstrap_ca.pem");
+  boost::filesystem::remove(conf.tls.certificates_directory / "bootstrap_cert.pem");
+  boost::filesystem::remove(conf.tls.certificates_directory / "bootstrap_pkey.pem");
```

Comment:
No big deal, but these bootstrap files aren't actually created anymore, so they don't need to be removed, right?